app/models.py

```
@@ -359,6 +351,62 @@ def prevent_class_timezone_mutation(_mapper, _connection, target):
+        db.CheckConstraint("economy_policy_mode IN ('tight', 'default', 'comfortable')", name='ck_economic_engine_mode'),
+        db.CheckConstraint('expected_weekly_hours IS NULL OR expected_weekly_hours > 0', name='ck_economic_engine_hours'),
+        db.CheckConstraint('interest_rate IS NULL OR (interest_rate >= 0 AND interest_rate <= 1.0)', name='ck_economic_engine_rate'),
+        db.CheckConstraint("interest_calculation_type IS NULL OR interest_calculation_type IN ('simple', 'compound')", name='ck_economic_engine_calc_type'),
+        db.CheckConstraint("compound_frequency IS NULL OR compound_frequency IN ('daily', 'weekly', 'monthly')", name='ck_economic_engine_compound_freq'),
+        db.CheckConstraint("interest_accrual_frequency IS NULL OR interest_accrual_frequency IN ('daily', 'weekly', 'monthly')", name='ck_economic_engine_accrual_freq'),
+        db.CheckConstraint("interest_payout_frequency IS NULL OR interest_payout_frequency IN ('weekly', 'monthly')", name='ck_economic_engine_payout_freq'),
+    )
+
+
+@event.listens_for(EconomicEngine, "before_update")
+def prevent_economic_engine_update(_mapper, _connection, target):
```

Ruff (0.16.1)

[warning] Unused function argument: `target`

(ARG001)